confirm.go

```diff
@@ -41,3 +41,37 @@ func NewConfirmCallbackWithTimeout(win fyne.Window, timeout time.Duration) func(
 		return <-resp
 	}
 }
+
+// NewRestartConfirmCallback create a GUI based confirmation to approve restarting the application after being updated
+func NewRestartConfirmCallback(win fyne.Window) func() bool {
+	return NewRestartConfirmCallbackWithTimeout(win, false, 0)
+}
+
+// NewRestartConfirmCallbackWithTimeout creates a GUI based restarting confirmation callback
+// with a timeout, after which time the question will be confirmed if defaultRestart is true.
+// This can assist in a "default to yes" update where computer may be unattended.
+func NewRestartConfirmCallbackWithTimeout(win fyne.Window, defaultRestart bool, timeout time.Duration) func() bool {
+	return func() bool {
+		var cancel func()
+		var d dialog.Dialog
+		resp := make(chan bool)
+		if timeout > 0 {
+			ctx, fn := context.WithTimeout(context.Background(), timeout)
+			cancel = fn
+			go func() {
+				<-ctx.Done()
+				d.Hide()
+				resp <- defaultRestart
+			}()
+		}
+		d = dialog.NewConfirm("Application Update", "Do you wish to restart this application?\n", func(ok bool) {
```

Comment:
We might need to give the user context here - they did not request the restart directly.
Something that mentions that the update is ready and the application needs to be restarted, is this OK.